File: 프로그래머스/1/12901. 2016년/2016년.js
function solution(a, b) {
    const calendar = {
  1: 31,
  2: 29,
  3: 31,
  4: 30,
  5: 31,
  6: 30,
  7: 31,
  8: 31,
  9: 30,
  10: 31,
  11: 30,
  12: 31,
};

const day = {
    0: "THU",
  1: "FRI",
  2: "SAT",
  3: "SUN",
  4: "MON",
  5: "TUE",
  6: "WED",
};
    let date = 0;
for (let i = 1; i < a; i++) {
  date += calendar[i];
}
date += b;
    let answer = date % 7;
    return day[answer];
}

File: 프로그래머스/1/12901. 2016년/2016년.java
import java.util.*;
class Solution {
    public String solution(int a, int b) {
        HashMap<Integer, Integer> calendar = new HashMap<>();
        HashMap<Integer, String> day = new HashMap<>();
        calendar.put(1, 31);
        calendar.put(2, 29);
        for(int i=3; i<=12; i++){
            if(i<=7){
                if(i%2!=0)calendar.put(i, 31);
                else calendar.put(i, 30);
            }else{
                if(i%2!=0) calendar.put(i, 30);
                else calendar.put(i, 31);
            }
        }
        day.put(0, "THU");
        day.put(1, "FRI");
        day.put(2, "SAT");
        day.put(3, "SUN");
        day.put(4, "MON");
        day.put(5, "TUE");
        day.put(6, "WED");
        int date = 0;
        for(int i=1; i<a; i++){
            date+=calendar.get(i);
        }
        date+=b;
        return day.get(date%7);
    }
}